src/test/librbd/object_map/test_mock_SnapshotCreateRequest.cc

```diff
@@ -130,7 +130,7 @@ TEST_F(TestMockObjectMapSnapshotCreateRequest, ReadMapError) {
     std::shared_lock image_locker{ictx->image_lock};
     request->send();
   }
-  ASSERT_EQ(0, cond_ctx.wait());
+  ASSERT_EQ(-ENOENT, cond_ctx.wait());
```

Comment:
Nit: you need to tweak the test to ensure the object map can be invalidated w/o an `-ENOENT`. Same comment applies below.

Reply:
Looks like this still needs to be addressed.

Reply:
@dillaman  Do you mean I should add tests w/o an `-ENOENT` in `expect_read_map`, and with `-ENOENT` in `expect_write_map` & `expect_add_snapshot`. Or would you help me how to do it? Thanks.

Reply:
I mean that these tests shouldn't be receiving `-ENOENT` errors, so something needs to get fixed.

Reply:
@dillaman  Fix by never propagate an `-ENOENT` back to the caller.

A doubt: it seems that validating object map didn't take effect after `send_read_map/send_write_map/send_add_snapshot` failed, because it will return `-ENOENT` at [image_ctx.get_flags](https://github.com/ceph/ceph/blob/master/src/librbd/object_map/InvalidateRequest.cc#L30).

Reply:
In the case where the snapshot object map was never created, we technically don't need to invalidate it since any attempt to load it will fail and mark it invalid at that point.

Reply:
Yeah, as you said, in the process of `object_map::SnapshotCreateRequest`, if read map fails, we technically don't need to invalidate it. So I think we expect an error (`ENOENT`) to be returned here is reasonable (as the tests name are ***Error and it always returns an `ENOENT` actually).
To be honest, I have no idea how to fix it, maybe passing `CEPH_NOSNAP` instead of `snap_id` at [SnapshotCreateRequest](https://github.com/ceph/ceph/pull/34059/commits/4d9aad06fedd57d84fa7c07b61214fba64230a19#diff-7561d125517840b2c7f07c4f7b817f4bR128) ? but I don't think it's reasonable.